### PR TITLE
Found challenge for day4.

### PR DESCRIPTION
**Pull request by sanddc318**

### CODE REVIEW

File: problems/sep_5_2018.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Using the JavaScript language, have the function LetterChanges(str) take the str parameter being passed and modify it using the following algorithm:
```
Comment (joe-tohdjojo) [inline problems/sep_5_2018.js]: try to break comments at column 80 for readability

File: problems/sep_5_2018.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+// Replace every letter in the string with the letter following it in the alphabet (ie. c becomes d, z becomes a). Then capitalize every vowel in this new string (a, e, i, o, u) and finally return this modified string.
+
+function LetterChanges(str) {
+  // code goes here
+  return str
+}
+
+// Do not change code below this line.
+LetterChanges(readline())
```
Comment (joe-tohdjojo) [inline problems/sep_5_2018.js]: is `readline` necessary?

**Review by joe-tohdjojo — APPROVED** [review]
There are 2 tasks for this challenge. Try to break up the tasks into 2 separate functions which you call in `LetterChanges`. Also, (bonus) try to make these functions pure. Meaning, don't mutate the original string but return a new string

File: problems/sep_5_2018.js [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
+// Test case:
+// 'hello*3 => Ifmmp*3
 
 function LetterChanges(str) {
-  // code goes here
-  return str
+  const letterShift = str.replace(/[a-z]/gi, (char) => {
+    return char === 'z' || char === 'Z'
+      ? 'a'
+      : String.fromCharCode(char.charCodeAt() + 1)
```
Comment (sanddc318) [inline problems/sep_5_2018.js]: I got line 16 from googling. I had no idea about the charCode methods. What I was doing was going nowhere.

File: problems/sep_5_2018.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+// Replace every letter in the string with the letter following it
+// in the alphabet (ie. c becomes d, z becomes a).
+// Then capitalize every vowel in this new string (a, e, i, o, u)
+// and finally return this modified string.
+
+// Test case:
+// 'hello*3 => Ifmmp*3
+
+function LetterChanges(str) {
```
Comment (joe-tohdjojo) [inline problems/sep_5_2018.js:12]: By convention, capital function names are reserved for classes

**Review by joe-tohdjojo — APPROVED** [review]
LGTM. this looks great. very clean

**Pull request merged**